File: app/src/main/java/com/blog/demo/component/intent/IntentCategoryActivity.java
package com.blog.demo.component.intent;

import android.app.Activity;
import android.os.Bundle;
import android.widget.TextView;

import androidx.annotation.Nullable;

import com.blog.demo.R;

public class IntentCategoryActivity extends Activity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_component_intent_category);

        TextView tvIntent = findViewById(R.id.tv_intent);
        tvIntent.setText(getIntent().getStringExtra("value"));
    }

}

File: app/src/main/java/com/blog/demo/component/intent/IntentCategoryActivity.kt
package com.blog.demo.component.intent

import android.app.Activity
import android.os.Bundle
import android.widget.TextView
import com.blog.demo.R

class IntentCategoryActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_component_intent_category)
        val tvIntent: TextView = findViewById(R.id.tv_intent)
        tvIntent.text = intent.getStringExtra("value")
    }

}